src/v/cloud_storage/partition_manifest.cc

```diff
@@ -1139,19 +1157,53 @@ void partition_manifest::serialize(std::ostream& out) const {
         }
         w.EndObject();
     };
+    auto serialize_lw_meta = [this, &w](const lw_segment_meta& meta) {
+        vassert(
+          meta.segment_term != model::term_id{},
+          "Term id is not initialized, base offset {}",
+          meta.base_offset);
+        auto sn = generate_local_segment_name(
+          meta.base_offset, meta.segment_term);
+        w.Key(sn());
+        w.StartObject();
+        w.Key("is_compacted");
+        w.Bool(true);
```

Comment:
I'm guessing it's because the fields just don't matter for the lw_segment_meta... although it feels a bit odd.

Since the _replaced list is a new part of the JSON format in 22.3, we could define it to just only have the fields we want for lw_segment meta